src/passes/Print.cpp

```diff
@@ -3627,15 +3623,18 @@ printStackIR(StackIR* ir, std::ostream& o, Function* func) {
       default:
         WASM_UNREACHABLE("unexpeted op");
     }
-    std::cout << '\n';
+    o << '\n';
   }
   assert(controlFlowDepth == 0);
   return o;
 }
 
-std::ostream& printStackIR(std::ostream& o, Module* module) {
+std::ostream& printStackIR(std::ostream& o, Module* module, bool optimize) {
   wasm::PassRunner runner(module);
   runner.add("generate-stack-ir");
+  if (optimize) {
+    runner.add("optimize-stack-ir");
+  }
```

Comment:
Regarding having `optimize` on API functions, perhaps here is why it makes sense: `printStackIR` already runs a pass (here: `generate-stack-ir`), so not just printing, and putting the optimize pass directly behind it seems good to do? The behavior is then made controllable via the C API functions calling `printStackIR`, with the side-effect of also matching what the JS API provides. On the AS side, we are always setting `optimize=true` to remove sequences of `unreachable`s and similar iirc.